### PR TITLE
SG-33870 Add methods for the user_subscriptions API end points

### PR DESCRIPTION
**Pull request by julien-lang**
The ShotGrid API3 recently introduced two new endpoints:

- user_subscriptions_create
- user_subscriptions_read

### CODE REVIEW

**Review by souphan-adsk — DISMISSED** [review]

**Review by NorberMV — DISMISSED** [review]
LGTM!

File: shotgun_api3/shotgun.py [hunk 1]
```diff
@@ -3261,6 +3261,42 @@ def preferences_read(self, prefs=None):
...
+
+    def user_subscriptions_create(self, users: list):
+        """
+        Assign subscriptions to users.
+
+        :param list users: list of user subscriptions to assign.
+            Each subscription must be a dictionary with the ``humanUserId`` and
+            ``subscription`` fields.
+            The ``subscription`` is either ``None``, a single string, or an
+            array of strings with subscription info that the user.
```
Comment (shaynacohen) [inline shotgun_api3/shotgun.py:3284]:
```suggestion
            array of strings with subscription information.
```

File: shotgun_api3/shotgun.py [hunk 2]
```diff
@@ -3261,6 +3261,42 @@ def preferences_read(self, prefs=None):
...
+
+        :returns: ``True`` if the request succedeed, ``False`` otherwise.
```
Comment (shaynacohen) [inline shotgun_api3/shotgun.py:3286]:
```suggestion
        :returns: ``True`` if the request succedeed, ``False`` if otherwise.
```

**Review by shaynacohen — APPROVED** [review]
@julien-lang added my suggestions!

**Review by NorberMV — APPROVED** [review]

**Pull request merged**